File: cplusplus/src/class1.cpp
#include "class1.hpp"

libimport int class1_add(int a, int b, int c);
libimport void class1_hello(void);

int class1::add(int a, int b, int c) {
    return class1_add(a,b,c);
}

void class1::hello(void){
    return class1_hello();
}

File: csharp/Class1.cs
using System;
using System.Runtime.InteropServices;

namespace CppCsComTest {

    public class Class1 {

        public static int Add(int a, int b, int c) {
            return Class2.Add( a, Class2.Add(b,c) );
        }

        public static void Hello() {
            Class2.Hello();
            Console.WriteLine("Hello from C# on " + DllHandle.OS + ".");
        }
        
    }

    public class ExportClass1 {

        [UnmanagedCallersOnly(EntryPoint = "class1_add")]
        public static int Add(int a, int b, int c) {
            return Class1.Add(a,b,c);
        }

        [UnmanagedCallersOnly(EntryPoint = "class1_hello")]
        public static void Hello() {
            Class1.Hello();
        }
        
    }

}